webapp-django/crashstats/supersearch/static/supersearch/js/socorro/search.js

```diff
@@ -3,144 +3,162 @@
 $(function () {
     'use strict';
 
-    // parameters
     var form = $('#search-form form');
-    var fieldsURL = form.data('fields-url');
     var resultsURL = form.data('results-url');
-    var customURL = form.data('custom-url');
-    var publicApiUrl = form.data('public-api-url');
-
-    var submitButton = $('button[type=submit]', form);
-    var newLineBtn = $('.new-line');
-    var customizeBtn = $('.customize');
+    var simpleSearchContainer = $('#simple-search');
 
     var contentElt = $('#search_results');
-    var optionsElt = $('fieldset.options', form);
     var sortInput = $('input[name=_sort]', form);
     var facetsInput = $('input[name=_facets]', form);
     var columnsInput = $('input[name=_columns_fake]', form);
-    var publicApiUrlInput = $('input[name=_public_api_url]', form);
 
-    function showResults(url) {
-        // Show loader.
-        try {
-            contentElt.tabs('destroy');
+    /**
+     * Run a search.
+     *
+     * Update the public API URL, push a new history state, query the search
+     * results and show them.
+     */
+    function search(noHistory, page) {
+        if (!page) {
+            page = 1;
         }
-        catch (e) {}
-        contentElt.empty().append($('<div>', {'class': 'loader'}));
 
-        // If a tracker is available, track that AJAX call.
-        Analytics.trackPageview(url);
+        var params = getParams();
+        updatePublicApiUrl(params);
 
-        $.ajax({
-            url: url,
-            success: function (data) {
-                contentElt.empty().append(data);
+        // Set the page after updating the public API URL, because we don't want
+        // that page parameter in the URL. It is only useful in the UI.
+        params.page = page;
 
-                if ($('.no-data', contentElt).length) {
-                    // There are no results, no need to do more.
-                    return;
-                }
+        var url = prepareResultsQueryString(params);
+        queryResults(resultsURL + url);
+
+        if (!noHistory) {
+            pushHistoryState(params, url);
+        }
+    }
+
+    /**
+     * Display results from a search query.
+     */
+    function showResults(data) {
+        contentElt.empty().append(data);
+
+        if ($('.no-data', contentElt).length) {
```

Comment:
This is odd. Can't you do an if statement on the `data` variable instead? Maybe not important.